src/arch/riscv/irqc/aia/inc/irqc.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _IRQC_H_
+#define _IRQC_H_
+
+#include <aplic.h>
+#include <cpu.h>
+#include <vaplic.h>
+
+#define XPLIC_MAX_INTERRUPTS (1024)
+
+#define HART_REG_OFF APLIC_IDC_OFF
+#define IRQC_HART_INST APLIC_PLAT_IDC_NUM
+
+static inline void irqc_init()
+{
+    aplic_init();
+}
+
+static inline void irqc_cpu_init()
+{
+    aplic_idc_init();
+}
+
+static inline void irqc_set_enbl(irqid_t int_id, bool en)
+{
+    aplic_set_ienum(int_id);
```

Comment:
My question, if we did that, would be the following. For an interrupt, x, to be correctly configured, it must first configure the sourcecfg[x] so that the interrupt becomes active in the domain. Only after that can it be activated. I leave the excerpt from the specification: "Whenever interrupt source i is inactive in an interrupt domain, the corresponding interrupt-pending and interrupt-enable bits within the domain are read-only zeros, and register target[i] is also read-only zero".
Setting the target register to a given default would involve checking if the register is not already written. I'm unsure if we should do that; Can you provide some insight into this?

Reply:
Yes, it was! It can be seen in [src/arch/riscv/irqc/aia/inc/irqc.h](https://github.com/bao-project/bao-hypervisor/pull/52/commits/61cf0895d828a1269c379c0dbca76d457304c1a3#diff-fb680f85e62c694862fbaa4b7a92e61d0f9348fd888d51dd4bfe0ff28c257c3c)